dataclasses_json/core.py

```diff
@@ -57,7 +57,7 @@ def default(self, o) -> Json:
 FieldOverride = namedtuple('FieldOverride', confs)
 
 
-@cached(cache=_user_overrides_cache, key=id)
+@cached(cache=_user_overrides_cache, key=repr)
```

Comment:
I am not quite sure why repr fixes the problem.
The default `__repr__` implementation returns basically `f"<{__name__}.{self.__class__.__name__} object at {id(self)}>`, so if the class is overwritten with the same name and the variable naming stuff is done as before, this will result in the same repr as well, or am I mistaken?

```
class Test():
...     pass
... 
Test().__repr__()
'<__main__.Test object at 0x7f458df2ab50>'
# Overwrite it
class Test():
...     pass
... 
Test().__repr__()
'<__main__.Test object at 0x7f458df2ab50>'
```

Reply:
Same for the other caches

Reply:
yeah true, I will try to find another solution

Reply:
fixed with an injected _dataclass_hash field. this now works correctly for all cases I can think of...